src/push_rules.rs

```diff
@@ -444,37 +444,47 @@ mod tests {
 
     #[test]
     fn serialize_string_action() {
-        assert_eq!(to_string(&Action::Notify).unwrap(), r#""notify""#);
+        assert_eq!(to_value(&Action::Notify).unwrap(), json!("notify"));
     }
 
     #[test]
     fn serialize_tweak_sound_action() {
         assert_eq!(
-            to_string(&Action::SetTweak(Tweak::Sound {
+            to_value(&Action::SetTweak(Tweak::Sound {
                 value: "default".to_string()
             }))
             .unwrap(),
-            r#"{"set_tweak":"sound","value":"default"}"#
+            json!({
+                "set_tweak": "sound",
+                "value": "default"
+            })
         );
     }
 
     #[test]
     fn serialize_tweak_highlight_action() {
         assert_eq!(
-            to_string(&Action::SetTweak(Tweak::Highlight { value: true })).unwrap(),
-            r#"{"set_tweak":"highlight","value":true}"#
+            to_value(&Action::SetTweak(Tweak::Highlight { value: true })).unwrap(),
+            json!({"set_tweak": "highlight", "value":true})
         );
     }
 
     #[test]
     fn deserialize_string_action() {
-        assert_eq!(from_str::<Action>(r#""notify""#).unwrap(), Action::Notify);
+        assert_eq!(
+            from_value::<Action>(json!("notify")).unwrap(),
+            Action::Notify
+        );
     }
 
     #[test]
     fn deserialize_tweak_sound_action() {
+        let json_data = json!({
+            "set_tweak": "sound",
+            "value": "default"
+        });
         assert_eq!(
-            from_str::<Action>(r#"{"set_tweak":"sound","value":"default"}"#).unwrap(),
+            from_value::<Action>(json_data).unwrap(),
```

Comment:
Feel free to remove the redundant generic arguments in this PR:

```suggestion
            from_value(json_data).unwrap(),
```

Reply:
ah it seems we need that type parameter
 ```
error[E0283]: type annotations needed
   --> src/push_rules.rs:489:13
    |
489 |             from_json_value(json_data).unwrap(),
    |             ^^^^^^^^^^^^^^^--------------------
    |             |
    |             this method call resolves to `T`
    |             cannot infer type for type parameter `T` declared on the function `from_value`
    |             help: consider specifying the type argument in the function call: `from_json_value::<T>`
```

Reply:
Ah, I thought it could be inferred here. Nevermind then.